redbot/core/utils/chat_formatting.py

```diff
@@ -283,6 +284,103 @@ def pagify(
+    always_page_number: bool = False,
+) -> List[discord.Embed]:
+    """
+    Parameters
+    ----------
+    rows : `list` of `str`
+        The strings that form a row in the embed
+    base_embed : `discord.Embed`
+        The embed to use as a base for the list of embeds
+    embed_max_fields : `int`, optional
+        The amount of fields each embed should have.
+        If ``greedy_fill`` is :code:`True`, the final embed may have fewer fields.
+    field_max_rows : `int`, optional
+        The maximum amount of rows per field.
+    footer_static : `str`, optional
+        The static text to add to the footer of each embed.
+        This text will appear after the page numbers.
+    greedy_fill : `bool`, optional
+        Whether to fill the fields greedily or not.
+        If :code:`True`, each field up to the last one will contain ``field_max_rows``,
+        where the last row will contain the remainder of rows.
+        If :code:`False`, all fields will contain an equal amount of rows,
+        with at most one additional remainder row.
+    always_page_number : `bool`, optional
+        Whether to always display page numbers in the footer, even with only one embed.
+        Defaults to :code:`False`.
+
+    Returns
+    -------
+    embed_list : `list` of `discord.Embed`
+        A list of embeds, based on the base embed, with all rows spread across the embeds.
+
+    Raises
+    ------
+    ValueError
+        When either ``embed_max_fields`` or ``field_max_rows`` is not positive.
+    """
+    if embed_max_fields < 1:
+        raise ValueError("An embed must contain at least 1 field.")
+    if field_max_rows < 1:
+        raise ValueError("A field must contain at least 1 row.")
+    row_cnt = len(rows)
+    embed_cnt = 1 + ((row_cnt - 1) // (field_max_rows * embed_max_fields))
+    if greedy_fill:  # Fill as much to the max length as possible.
+        field_quot, row_remainder = divmod(row_cnt, field_max_rows)
+        field_sizes: List[int] = [field_max_rows] * field_quot
+        if row_remainder > 0:  # Check if remainder row necessary.
+            field_sizes.append(row_remainder)
+    else:  # Spread as equally as possible.
+        field_cnt = embed_cnt * embed_max_fields
+        # Check how many rows needed per field.
+        row_quot, row_remainder = divmod(row_cnt, field_cnt)
+        # Equally distribute rows over fields, with at most 1 extra row per field.
+        # The extra rows are filled from beginning to end.
+        field_sizes = [row_quot + int(i < row_remainder) for i in range(field_cnt)]
+    # Initialise footer format.
+    foot_str = None
+    if always_page_number or embed_cnt > 1:
+        foot_str = "{_page} of {_total}"  # _a and _z to be filled in the loop.
+        if footer_static:
+            foot_str += " — {}".format(footer_static)
+    elif footer_static:
+        foot_str = footer_static
+    # Initialise for nested loop.
+    embed_list: List[discord.Embed] = []
+    start = 0
+    f = 0
```

Comment:
This doesn't appear to be used anywhere

Reply:
- `embed_list` is used in L380 (to append)
- `start` is used several times (lines 371, 373m 376)
- `f` is incremented in L379

Reply:
I meant specifically f, it's increments but I don't see it being used anywhere other then in the line which it gets incremented